src/Raven.Server/Utils/Cpu/CpuUsageExtensionPoint.cs

```diff
@@ -43,124 +42,97 @@ public class CpuUsageExtensionPoint : IDisposable
-                    {
-                        break;
+                        var lineOutHandler = new Action<object, string>((p, l) =>
+                        {
+                            Interlocked.Exchange(ref _isDataValid, 1);
+
+                            if (l == null)
+                            {
+                                if (DateTime.UtcNow - lastReceivedLine > TimeSpan.FromSeconds(60))
+                                {
+                                    Interlocked.Exchange(ref _isDataValid, 0);
+                                    NotifyWarning("Cpu usage process hanged (no output for 60 seconds), killing the process",
+                                        new TimeoutException("no output from process for 60 seconds"));
+                                    cts.Cancel();
+                                }
+
+                                return;
+                            }
+
+                            lastReceivedLine = DateTime.UtcNow;
+                            var errString = HandleInfoReceived(l);
+                            if (errString != null)
+                            {
+                                Interlocked.Exchange(ref _isDataValid, 0);
+                                NotifyWarning(errString);
+                                cts.Cancel();
+                            }
+                        });
+
+                        RavenProcess.Execute(
+                            _startInfo.FileName,
+                            _startInfo.Arguments,
+                            1,
+                            null,
+                            lineOutHandler,
+                            linkedCts.Token);
+
+                        Interlocked.Exchange(ref _isDataValid, 0);
+
+                        if (ctk.IsCancellationRequested)
+                            return;
+
+                        if (DateTime.UtcNow - lastRestart > TimeSpan.FromMinutes(maxMinutesBetweenIssues))
+                        {
+                            retry = 2;
+                            lastRestart = DateTime.UtcNow;
+                            NotifyWarning($"'Restarting '{_startInfo.FileName} {_startInfo.Arguments}'.", new TimeoutException("Cpu usage process restart"));
+                        }
                     }
                 }
+
+                NotifyWarning(
+                        $"Failed twice in {maxMinutesBetweenIssues} minutes to get cpu usage vi extension point process withing and therefore the process will terminate.",
+                        new TimeoutException("Cpu usage process terminate"));
             }
             catch (Exception e)
             {
-                HandleError(e.Message, e);
+                NotifyWarning("Could not start cpu usage extension point process", e);
+                // ignore
             }
         }
 
-        private bool HandleInfoReceived(string data)
+        private string HandleInfoReceived(string data)
```

Comment:
Maybe `bool TryHandleInfoReceived(string data, out string error)` instead to be more explicit if the method succeeds or not. Returning `null` on success isn't obvious and can be a surprise for someone in the future